src/main/proto/analysis.proto

```diff
@@ -0,0 +1,41 @@
+/* Description of analyzers, predefined and custom */
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.apache.platypus.server.grpc";
+option java_outer_classname = "AnalysisProto";
+option objc_class_prefix = "HLW";
+
+package luceneserver;
+
+message NameAndParams {
+    string name = 1;
+    map<string, string> params = 2;
+}
+
+message ConditionalTokenFilter {
+    NameAndParams condition = 1;
+    repeated NameAndParams tokenFilters = 2;
+}
+
+// Used to be able to check if a value was set
+message IntObject {
+    int32 int = 1;
+}
+
+message CustomAnalyzer {
+    repeated NameAndParams charFilters = 1; // Available char filters as of Lucene 8.2.0: htmlstrip, mapping, persian, patternreplace
+    NameAndParams tokenizer = 2;
+    repeated NameAndParams tokenFilters = 3;
```

Comment:
probably comment pointing to org.apache.lucene.analysis.TokenFilter and stating the possible tokenFilters to be used?

Reply:
There are 101 possible token filters, can't write each one of them. I added links to the Lucene pages though. We can probably have a separate page for listing the token filters in the future if needed.